sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/rx/QueryValidationTests.java

```diff
@@ -317,6 +325,132 @@ public void queryPlanCacheSinglePartitionParameterizedQueriesCorrectness() {
 
     }
 
+    @Test(groups = {"simple"}, timeOut = TIMEOUT * 10)
```

Comment:
simulating partition split state as integration test is not easy.

I wonder if that's possible to add a mocking unit test for this?
Please just think about it. if It is not easy then ignore.

Reply:
I tried mocking options, but finally decided it is more convenient to do a live test than mock. I have some ideas for mocking and will try that too but for now live test should probably be good?

Reply:
Consider adding tests like this:

[https://github.com/Azure/azure-cosmos-dotnet-v3/blob/master/Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Query/ContinuationResumeLogicTests.cs](https://github.com/Azure/azure-cosmos-dotnet-v3/blob/master/Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/Query/ContinuationResumeLogicTests.cs)

---
In reply to: [561152747](https://github.com/Azure/azure-sdk-for-java/pull/18698#discussion_r561152747) [](ancestors = 561152747)